SCRUM-182-new: add comment to post

eliseevais opened the pull request.

doroshkent committed:
update textfield to handle search type

doroshkent committed:
Add users API integration and debounce hook

Introduce `usersApi` with support for fetching users via query parameters such as search, pagination, and cursors. Define corresponding TypeScript types for API responses and arguments. Additionally, add a reusable `useDebounce` hook for managing debounced state updates.

doroshkent committed:
Update Avatar with medium size and enhance Scroll component

Added a "medium" size option to the Avatar component, along with corresponding styles. Refactored the Scroll component to support custom classNames, a scroll event handler, and implemented `forwardRef` for better flexibility.

doroshkent committed:
Add user profile queries and follow/unfollow mutations

Introduced `getUser` query, as well as `followUser` and `unfollowUser` mutations to the `usersApi`. Updated associated types to support fetching user profiles and managing follow/unfollow actions. These changes enhance the API with user interaction capabilities.

eliseevais committed:
SCRUM-182-new: add comment to post

eliseevais committed:
SCRUM-162: add capitalizeFirstLetter function

doroshkent commented on src/common/hooks/useHandleAddComment.ts:
```diff
@@ -0,0 +1,15 @@
+    try {
+      await createComment({ postId, content: commentText }).unwrap();
+    } catch (error) {
+      console.error("Error while creating comment:", error);
```
you've mentioned, it is better to avoid console logs

doroshkent committed:
Prevent duplicate triggers in the search page scroll logic.

Added `isLoading` to the dependency array of the `useCallback` hook to ensure the scroll logic does not trigger while a fetch is in progress. This prevents potential race conditions and improves performance.

doroshkent committed:
Refactor and consolidate MyPost and Profile components

Removed `myPost` components and consolidated post-related functionality into common components. Streamlined `Profile` structure by integrating reusable components, improving modularity. Adjusted API types (`Post` -> `PostData`) to align with updated design.

eliseevais committed:
remove console.log

eliseevais committed:
SCRUM-182-new-new: add new comment form
